src/main/java/uk/gov/digital/ho/hocs/application/aspects/MessageHandlerContextAspect.java

```diff
@@ -7,35 +7,36 @@
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 import uk.gov.digital.ho.hocs.application.ClientContext;
+import uk.gov.digital.ho.hocs.domain.model.Message;
 
 @Component
 @Aspect
-public class QueueListenerContextAspect {
+public class MessageHandlerContextAspect {
 
     private final ClientContext clientContext;
     private final String user;
     private final String group;
     private final String team;
 
-    public QueueListenerContextAspect(ClientContext clientContext,
-                               @Value("${case.creator.identity.user}") String user,
-                               @Value("${case.creator.identity.group}") String group,
-                               @Value("${case.creator.identity.team}") String team) {
+    public MessageHandlerContextAspect(ClientContext clientContext,
+                                       @Value("${case.creator.identity.user}") String user,
+                                       @Value("${case.creator.identity.group}") String group,
+                                       @Value("${case.creator.identity.team}") String team) {
         this.clientContext = clientContext;
         this.user = user;
         this.group = group;
         this.team = team;
     }
 
-    @Pointcut("execution(* uk.gov.digital.ho.hocs.entrypoint.QueueListener.onMessageReceived(..))")
+    @Pointcut("execution(* uk.gov.digital.ho.hocs.domain.queue.common.MessageHandler.handleMessage(..))")
     public void messageReceivedPointcut(){
     }
 
     @Around("messageReceivedPointcut()")
     public void aroundAdvice(ProceedingJoinPoint joinPoint) throws Throwable {
         var getArguments = joinPoint.getArgs();
-        var messageId = (String) getArguments[1];
-        clientContext.setContext(user, group, team, messageId);
+        var message = (Message) getArguments[0];
+        clientContext.setContext(user, group, team, message.id());
         joinPoint.proceed();
```

Comment:
We need a way of returning the old values to the client context after the proceed here.